File: binding/include/charts/libqbarset.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qbarset.html
pub const qbarset = struct {

    /// New constructs a new QBarSet object.
    pub fn New(label: []const u8) QtC.QBarSet {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
return qtc.QBarSet_new(label_str);
    }


    /// New2 constructs a new QBarSet object.
    pub fn New2(label: []const u8, parent: ?*anyopaque) QtC.QBarSet {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
return qtc.QBarSet_new2(label_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarSet_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarset.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetLabel(self: ?*anyopaque, label: []const u8) void {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
qtc.QBarSet_SetLabel(@ptrCast(self), label_str);
    }

    pub fn Label(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarSet_Label(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarset.Label: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Append(self: ?*anyopaque, value: f64) void {
        qtc.QBarSet_Append(@ptrCast(self), value);
    }

    pub fn Append2(self: ?*anyopaque, values: []const u8) void {
        qtc.QBarSet_Append2(@ptrCast(self), @ptrCast(values));
    }

    pub fn OperatorShiftLeft(self: ?*anyopaque, value: *const f64) QtC.QBarSet {
        return qtc.QBarSet_OperatorShiftLeft(@ptrCast(self), @ptrCast(value));
    }

    pub fn Insert(self: ?*anyopaque, index: i32, value: f64) void {
        qtc.QBarSet_Insert(@ptrCast(self), index, value);
    }

    pub fn Remove(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_Remove(@ptrCast(self), index);
    }

    pub fn Replace(self: ?*anyopaque, index: i32, value: f64) void {
        qtc.QBarSet_Replace(@ptrCast(self), index, value);
    }

    pub fn At(self: ?*anyopaque, index: i32) f64 {
        return qtc.QBarSet_At(@ptrCast(self), index);
    }

    pub fn OperatorSubscript(self: ?*anyopaque, index: i32) f64 {
        return qtc.QBarSet_OperatorSubscript(@ptrCast(self), index);
    }

    pub fn Count(self: ?*anyopaque, ) i32 {
        return qtc.QBarSet_Count(@ptrCast(self));
    }

    pub fn Sum(self: ?*anyopaque, ) f64 {
        return qtc.QBarSet_Sum(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QBarSet_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QBarSet_Pen(@ptrCast(self));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QBarSet_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QBarSet_Brush(@ptrCast(self));
    }

    pub fn SetLabelBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QBarSet_SetLabelBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn LabelBrush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QBarSet_LabelBrush(@ptrCast(self));
    }

    pub fn SetLabelFont(self: ?*anyopaque, font: ?*anyopaque) void {
        qtc.QBarSet_SetLabelFont(@ptrCast(self), @ptrCast(font));
    }

    pub fn LabelFont(self: ?*anyopaque, ) QtC.QFont {
        return qtc.QBarSet_LabelFont(@ptrCast(self));
    }

    pub fn Color(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QBarSet_Color(@ptrCast(self));
    }

    pub fn SetColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_SetColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn BorderColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QBarSet_BorderColor(@ptrCast(self));
    }

    pub fn SetBorderColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_SetBorderColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn LabelColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QBarSet_LabelColor(@ptrCast(self));
    }

    pub fn SetLabelColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_SetLabelColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn SelectedColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QBarSet_SelectedColor(@ptrCast(self));
    }

    pub fn SetSelectedColor(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QBarSet_SetSelectedColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn IsBarSelected(self: ?*anyopaque, index: i32) bool {
        return qtc.QBarSet_IsBarSelected(@ptrCast(self), index);
    }

    pub fn SelectBar(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_SelectBar(@ptrCast(self), index);
    }

    pub fn DeselectBar(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_DeselectBar(@ptrCast(self), index);
    }

    pub fn SetBarSelected(self: ?*anyopaque, index: i32, selected: bool) void {
        qtc.QBarSet_SetBarSelected(@ptrCast(self), index, selected);
    }

    pub fn SelectAllBars(self: ?*anyopaque, ) void {
        qtc.QBarSet_SelectAllBars(@ptrCast(self));
    }

    pub fn DeselectAllBars(self: ?*anyopaque, ) void {
        qtc.QBarSet_DeselectAllBars(@ptrCast(self));
    }

    pub fn SelectBars(self: ?*anyopaque, indexes: []const u8) void {
        qtc.QBarSet_SelectBars(@ptrCast(self), @ptrCast(indexes));
    }

    pub fn DeselectBars(self: ?*anyopaque, indexes: []const u8) void {
        qtc.QBarSet_DeselectBars(@ptrCast(self), @ptrCast(indexes));
    }

    pub fn ToggleSelection(self: ?*anyopaque, indexes: []const u8) void {
        qtc.QBarSet_ToggleSelection(@ptrCast(self), @ptrCast(indexes));
    }

    pub fn SelectedBars(self: ?*anyopaque, ) []const u8 {
        return qtc.QBarSet_SelectedBars(@ptrCast(self));
    }

    pub fn Clicked(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_Clicked(@ptrCast(self), index);
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, status: bool, index: i32) void {
        qtc.QBarSet_Hovered(@ptrCast(self), status, index);
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Pressed(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_Pressed(@ptrCast(self), index);
    }

    pub fn OnPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_Pressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Released(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_Released(@ptrCast(self), index);
    }

    pub fn OnReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_Released(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DoubleClicked(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_DoubleClicked(@ptrCast(self), index);
    }

    pub fn OnDoubleClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_DoubleClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PenChanged(self: ?*anyopaque, ) void {
        qtc.QBarSet_PenChanged(@ptrCast(self));
    }

    pub fn OnPenChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_PenChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BrushChanged(self: ?*anyopaque, ) void {
        qtc.QBarSet_BrushChanged(@ptrCast(self));
    }

    pub fn OnBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_BrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelChanged(self: ?*anyopaque, ) void {
        qtc.QBarSet_LabelChanged(@ptrCast(self));
    }

    pub fn OnLabelChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_LabelChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelBrushChanged(self: ?*anyopaque, ) void {
        qtc.QBarSet_LabelBrushChanged(@ptrCast(self));
    }

    pub fn OnLabelBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_LabelBrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelFontChanged(self: ?*anyopaque, ) void {
        qtc.QBarSet_LabelFontChanged(@ptrCast(self));
    }

    pub fn OnLabelFontChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_LabelFontChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_ColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QBarSet_Connect_ColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BorderColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_BorderColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnBorderColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QBarSet_Connect_BorderColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QBarSet_LabelColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnLabelColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QBarSet_Connect_LabelColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SelectedColorChanged(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QBarSet_SelectedColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnSelectedColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QBarSet_Connect_SelectedColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ValuesAdded(self: ?*anyopaque, index: i32, count: i32) void {
        qtc.QBarSet_ValuesAdded(@ptrCast(self), index, count);
    }

    pub fn OnValuesAdded(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_ValuesAdded(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ValuesRemoved(self: ?*anyopaque, index: i32, count: i32) void {
        qtc.QBarSet_ValuesRemoved(@ptrCast(self), index, count);
    }

    pub fn OnValuesRemoved(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_ValuesRemoved(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ValueChanged(self: ?*anyopaque, index: i32) void {
        qtc.QBarSet_ValueChanged(@ptrCast(self), index);
    }

    pub fn OnValueChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QBarSet_Connect_ValueChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SelectedBarsChanged(self: ?*anyopaque, indexes: []const u8) void {
        qtc.QBarSet_SelectedBarsChanged(@ptrCast(self), @ptrCast(indexes));
    }

    pub fn OnSelectedBarsChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QBarSet_Connect_SelectedBarsChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarSet_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarset.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarSet_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarset.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Remove2(self: ?*anyopaque, index: i32, count: i32) void {
        qtc.QBarSet_Remove2(@ptrCast(self), index, count);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QBarSet_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqbarset.cpp
#include <QBarSet>
#include <QBrush>
#include <QColor>
#include <QFont>
#include <QObject>
#include <QPen>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qbarset.h>
#include "libqbarset.h"
#include "libqbarset.hxx"

QBarSet* QBarSet_new(const libqt_string label) {
	 return new QBarSet(QString::fromUtf8(label.data, label.len));
}

QBarSet* QBarSet_new2(const libqt_string label, QObject* parent) {
	 return new QBarSet(QString::fromUtf8(label.data, label.len), parent);
}

libqt_string QBarSet_Tr(const char* s) {
	QString _ret = QBarSet::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QBarSet_SetLabel(QBarSet* self, const libqt_string label) {
	self->setLabel(QString::fromUtf8(label.data, label.len));
}

libqt_string QBarSet_Label(const QBarSet* self) {
	QString _ret = self->label();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QBarSet_Append(QBarSet* self, const double value) {
	self->append(value);
}

void QBarSet_Append2(QBarSet* self, const libqt_list values) {
	self->append(QList<double>());
}

QBarSet* QBarSet_OperatorShiftLeft(QBarSet* self, const double* value) {
	return new QBarSet(self->operator<<(*value));
}

void QBarSet_Insert(QBarSet* self, const int index, const double value) {
	self->insert(index, value);
}

void QBarSet_Remove(QBarSet* self, const int index) {
	self->remove(index);
}

void QBarSet_Replace(QBarSet* self, const int index, const double value) {
	self->replace(index, value);
}

double QBarSet_At(const QBarSet* self, const int index) {
	return self->at(index);
}

double QBarSet_OperatorSubscript(const QBarSet* self, const int index) {
	return self->operator[](index);
}

int QBarSet_Count(const QBarSet* self) {
	return self->count();
}

double QBarSet_Sum(const QBarSet* self) {
	return self->sum();
}

void QBarSet_SetPen(QBarSet* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QBarSet_Pen(const QBarSet* self) {
	return new QPen(self->pen());
}

void QBarSet_SetBrush(QBarSet* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QBarSet_Brush(const QBarSet* self) {
	return new QBrush(self->brush());
}

void QBarSet_SetLabelBrush(QBarSet* self, const QBrush* brush) {
	self->setLabelBrush(*brush);
}

QBrush* QBarSet_LabelBrush(const QBarSet* self) {
	return new QBrush(self->labelBrush());
}

void QBarSet_SetLabelFont(QBarSet* self, const QFont* font) {
	self->setLabelFont(*font);
}

QFont* QBarSet_LabelFont(const QBarSet* self) {
	return new QFont(self->labelFont());
}

QColor* QBarSet_Color(QBarSet* self) {
	return new QColor(self->color());
}

void QBarSet_SetColor(QBarSet* self, QColor* color) {
	self->setColor(*color);
}

QColor* QBarSet_BorderColor(QBarSet* self) {
	return new QColor(self->borderColor());
}

void QBarSet_SetBorderColor(QBarSet* self, QColor* color) {
	self->setBorderColor(*color);
}

QColor* QBarSet_LabelColor(QBarSet* self) {
	return new QColor(self->labelColor());
}

void QBarSet_SetLabelColor(QBarSet* self, QColor* color) {
	self->setLabelColor(*color);
}

QColor* QBarSet_SelectedColor(const QBarSet* self) {
	return new QColor(self->selectedColor());
}

void QBarSet_SetSelectedColor(QBarSet* self, const QColor* color) {
	self->setSelectedColor(*color);
}

bool QBarSet_IsBarSelected(const QBarSet* self, int index) {
	return self->isBarSelected(index);
}

void QBarSet_SelectBar(QBarSet* self, int index) {
	self->selectBar(index);
}

void QBarSet_DeselectBar(QBarSet* self, int index) {
	self->deselectBar(index);
}

void QBarSet_SetBarSelected(QBarSet* self, int index, bool selected) {
	self->setBarSelected(index, selected);
}

void QBarSet_SelectAllBars(QBarSet* self) {
	self->selectAllBars();
}

void QBarSet_DeselectAllBars(QBarSet* self) {
	self->deselectAllBars();
}

void QBarSet_SelectBars(QBarSet* self, const libqt_list indexes) {
	self->selectBars(QList<int>());
}

void QBarSet_DeselectBars(QBarSet* self, const libqt_list indexes) {
	self->deselectBars(QList<int>());
}

void QBarSet_ToggleSelection(QBarSet* self, const libqt_list indexes) {
	self->toggleSelection(QList<int>());
}

libqt_list QBarSet_SelectedBars(const QBarSet* self) {
	auto _ret = self->selectedBars();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		auto& _elem = _ret[_i];
		_data[_i] = new std::remove_reference_t<decltype(_elem)>(_elem);
	}
	return _arr;
}

void QBarSet_Clicked(QBarSet* self, int index) {
	self->clicked(index);
}

void QBarSet_Connect_Clicked(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int) = reinterpret_cast<void (*)(QBarSet*, int)>(slot);
    QBarSet::connect(self, &QBarSet::clicked, [self, slotFunc](int index) {
	slotFunc(self, index);
    });
}

void QBarSet_Hovered(QBarSet* self, bool status, int index) {
	self->hovered(status, index);
}

void QBarSet_Connect_Hovered(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, bool, int) = reinterpret_cast<void (*)(QBarSet*, bool, int)>(slot);
    QBarSet::connect(self, &QBarSet::hovered, [self, slotFunc](bool status, int index) {
	slotFunc(self, status, index);
    });
}

void QBarSet_Pressed(QBarSet* self, int index) {
	self->pressed(index);
}

void QBarSet_Connect_Pressed(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int) = reinterpret_cast<void (*)(QBarSet*, int)>(slot);
    QBarSet::connect(self, &QBarSet::pressed, [self, slotFunc](int index) {
	slotFunc(self, index);
    });
}

void QBarSet_Released(QBarSet* self, int index) {
	self->released(index);
}

void QBarSet_Connect_Released(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int) = reinterpret_cast<void (*)(QBarSet*, int)>(slot);
    QBarSet::connect(self, &QBarSet::released, [self, slotFunc](int index) {
	slotFunc(self, index);
    });
}

void QBarSet_DoubleClicked(QBarSet* self, int index) {
	self->doubleClicked(index);
}

void QBarSet_Connect_DoubleClicked(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int) = reinterpret_cast<void (*)(QBarSet*, int)>(slot);
    QBarSet::connect(self, &QBarSet::doubleClicked, [self, slotFunc](int index) {
	slotFunc(self, index);
    });
}

void QBarSet_PenChanged(QBarSet* self) {
	self->penChanged();
}

void QBarSet_Connect_PenChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*) = reinterpret_cast<void (*)(QBarSet*)>(slot);
    QBarSet::connect(self, &QBarSet::penChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QBarSet_BrushChanged(QBarSet* self) {
	self->brushChanged();
}

void QBarSet_Connect_BrushChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*) = reinterpret_cast<void (*)(QBarSet*)>(slot);
    QBarSet::connect(self, &QBarSet::brushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QBarSet_LabelChanged(QBarSet* self) {
	self->labelChanged();
}

void QBarSet_Connect_LabelChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*) = reinterpret_cast<void (*)(QBarSet*)>(slot);
    QBarSet::connect(self, &QBarSet::labelChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QBarSet_LabelBrushChanged(QBarSet* self) {
	self->labelBrushChanged();
}

void QBarSet_Connect_LabelBrushChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*) = reinterpret_cast<void (*)(QBarSet*)>(slot);
    QBarSet::connect(self, &QBarSet::labelBrushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QBarSet_LabelFontChanged(QBarSet* self) {
	self->labelFontChanged();
}

void QBarSet_Connect_LabelFontChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*) = reinterpret_cast<void (*)(QBarSet*)>(slot);
    QBarSet::connect(self, &QBarSet::labelFontChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QBarSet_ColorChanged(QBarSet* self, QColor* color) {
	self->colorChanged(*color);
}

void QBarSet_Connect_ColorChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, QColor*) = reinterpret_cast<void (*)(QBarSet*, QColor*)>(slot);
    QBarSet::connect(self, &QBarSet::colorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QBarSet_BorderColorChanged(QBarSet* self, QColor* color) {
	self->borderColorChanged(*color);
}

void QBarSet_Connect_BorderColorChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, QColor*) = reinterpret_cast<void (*)(QBarSet*, QColor*)>(slot);
    QBarSet::connect(self, &QBarSet::borderColorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QBarSet_LabelColorChanged(QBarSet* self, QColor* color) {
	self->labelColorChanged(*color);
}

void QBarSet_Connect_LabelColorChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, QColor*) = reinterpret_cast<void (*)(QBarSet*, QColor*)>(slot);
    QBarSet::connect(self, &QBarSet::labelColorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QBarSet_SelectedColorChanged(QBarSet* self, const QColor* color) {
	self->selectedColorChanged(*color);
}

void QBarSet_Connect_SelectedColorChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, const QColor*) = reinterpret_cast<void (*)(QBarSet*, const QColor*)>(slot);
    QBarSet::connect(self, &QBarSet::selectedColorChanged, [self, slotFunc](const QColor& color) {
	slotFunc(self, color);
    });
}

void QBarSet_ValuesAdded(QBarSet* self, int index, int count) {
	self->valuesAdded(index, count);
}

void QBarSet_Connect_ValuesAdded(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int, int) = reinterpret_cast<void (*)(QBarSet*, int, int)>(slot);
    QBarSet::connect(self, &QBarSet::valuesAdded, [self, slotFunc](int index, int count) {
	slotFunc(self, index, count);
    });
}

void QBarSet_ValuesRemoved(QBarSet* self, int index, int count) {
	self->valuesRemoved(index, count);
}

void QBarSet_Connect_ValuesRemoved(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int, int) = reinterpret_cast<void (*)(QBarSet*, int, int)>(slot);
    QBarSet::connect(self, &QBarSet::valuesRemoved, [self, slotFunc](int index, int count) {
	slotFunc(self, index, count);
    });
}

void QBarSet_ValueChanged(QBarSet* self, int index) {
	self->valueChanged(index);
}

void QBarSet_Connect_ValueChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, int) = reinterpret_cast<void (*)(QBarSet*, int)>(slot);
    QBarSet::connect(self, &QBarSet::valueChanged, [self, slotFunc](int index) {
	slotFunc(self, index);
    });
}

void QBarSet_SelectedBarsChanged(QBarSet* self, const libqt_list indexes) {
	self->selectedBarsChanged(QList<int>());
}

void QBarSet_Connect_SelectedBarsChanged(QBarSet* self, intptr_t slot) {
    void (*slotFunc)(QBarSet*, libqt_list) = reinterpret_cast<void (*)(QBarSet*, libqt_list)>(slot);
    QBarSet::connect(self, &QBarSet::selectedBarsChanged, [self, slotFunc](const QList<int>& indexes) {
	slotFunc(self, indexes);
    });
}

libqt_string QBarSet_Tr2(const char* s, const char* c) {
	QString _ret = QBarSet::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QBarSet_Tr3(const char* s, const char* c, int n) {
	QString _ret = QBarSet::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QBarSet_Remove2(QBarSet* self, const int index, const int count) {
	self->remove(index, count);
}

void QBarSet_Delete(QBarSet* self) {
    delete self;
}

